lib/cli/src/Cardano/CLI.hs

```diff
@@ -1576,26 +1597,85 @@ mkScribeId :: LogOutput -> ScribeId
 mkScribeId (LogToStdout _) = "StdoutSK::text"
 mkScribeId (LogToFile file _) = T.pack $ "FileSK::" <> file
 
+getPrometheusURL :: IO (Maybe (String, Port "Prometheus"))
+getPrometheusURL = do
+    prometheus_port <- lookupEnv "CARDANO_WALLET_PROMETHEUS_PORT"
+    prometheus_host <- lookupEnv "CARDANO_WALLET_PROMETHEUS_HOST"
+    case (prometheus_host, prometheus_port) of
+        (Just host, Just port) -> do
+            let parsedHost = readMaybe @String host
+                parsedPort = fromText @(Port "Prometheus") $ T.pack port
+            case (parsedHost, parsedPort) of
+                (Just host', Right port') -> pure $ Just (host', port')
+                _ -> do
+                    TIO.hPutStr stderr
+                        "Host or port value for prometheus metrics invalid. Will be disabled."
+                    pure Nothing
+        _ -> pure Nothing
+
+getEKGURL :: IO (Maybe (String, Port "EKG"))
+getEKGURL = do
+    ekg_port <- lookupEnv "CARDANO_WALLET_EKG_PORT"
+    ekg_host <- lookupEnv "CARDANO_WALLET_EKG_HOST"
+    case (ekg_host, ekg_port) of
+        (Just host, Just port) -> do
+            let parsedHost = readMaybe @String host
```

Comment:
`readMaybe @String` expects surrounding `""`, so I'd take you simply want `host`.